docs/Custom_Device_Types.md

```diff
@@ -0,0 +1,184 @@
+
+![](images/Read_Data_from_HW_State.jpg)
+
+**Figure: Read Data from HW State of the Inline Hardware Interface Framework**
+ 
+The template contains a Flat Sequence frame named Read Hardware Channels. You can replace the code inside the frame with the API calls necessary to obtain data from a hardware API.
+
+Do not call **Get** or **Set Channel Value by Data Reference** outside the inline driver VI.
+Doing so could cause system instability or errors.
+
+### Write Data to HW
+
+The Write Data to HW case executes at the end of the PCL, after the other components (such as stimulus profiles, faults, alarms, procedures, etc. ) have executed.
+
+![](images/Write_Data_to_HW_State.jpg)
+
+**Figure: Write Data to HW State of the Inline Hardware Interface Framework**
+
+The case contains a Flat Sequence frame named Write Input Data to Hardware Channels. You can replace the code inside the frame with the API calls necessary to send data to a hardware device.
+
+### Close
+
+The Close case executes after the PCL has finished executing. It's good practice to close references and release resources in this state. Since the PCL has terminated, you cannot read or write channel values in this case.
+
+### Inline Model Interface
+
+The Inline Model Interface custom device template is state machine/action engine architecture. An uninitialized Feedback Node is used for iterative data transfer. There are four cases defined by the Operation enumerated control.
+
+1.	Initialize – Same as Inline HW Interface
+2.	Start – Same as Inline HW Interface
+3.	Execute Model
+4.	Close – Same as Inline HW Interface
+
+This custom device is run in-line with the PCL, which calls each case at a specific time with respect to the other components in the system. The PCL will not proceed until the custom device case has returned.
+
+### Execute Model
+
+The execute model case is called in the middle of the PCL. This is the one state of this device that executes during the PCL. This state reads input data, performs a calculation, and then writes output data to VeriStand. Using the Inline Model Interface mode enables you to process data acquired from hardware inputs and send the processed values to hardware outputs with no latency.
+
+![](images/Execute_Model_State.jpg)
+
+**Figure: Execute Model State of the Inline Model Interface Framework**
+
+Do not call **Get** or **Set Channel Value by Data Reference** outside the inline driver VI. Doing so could cause system instability or errors.
+ 
+### Pre-defined Custom Device Types
+
+The type of a custom device refers to its execution mode, which defines how the device interacts with the VeriStand Engine.
+
+The **[following table](https://www.ni.com/documentation/en/veristand/latest/manual/custom-device-types/)** displays the pre-defined custom device types that are included in VeriStand.
```

Comment:
please fix the wording "The following table" as the table is not following this statement - rather the link navigates you to a table in the online product help.  something like "This table" at the link text would be an improvement?